Homework/Homework5/Services/LoggerService.cs

```diff
@@ -0,0 +1,83 @@
+﻿using System.Text;
+using Homework5;
+
+class Logger
+{
+    private static Logger? instance;
+    private string received_log = "";
+    private readonly string logFile = "log.txt";
+
+    private Logger() { }
+
+    public static Logger Instance
+    {
+        get
+        {
+            instance ??= new Logger();
+            return instance;
+        }
+    }
+
+    public void Log(LogLevel logLevel, string logMessage)
+    {
+        ChangeLogColor(logLevel);
+
+        string logTime = DateTime.Now.ToString("yyyy-MM-dd HH:mm:ss.fff");
+        string formattedLog = FormatLog(logTime, logLevel, logMessage);
+
+        Console.WriteLine(formattedLog);
+        received_log += formattedLog + "\n";
+
+        Console.ForegroundColor = ConsoleColor.White;
+    }
+
+    public void WriteLogsWithConfirmation()
+    {
+        ConsoleKeyInfo confirmationResult;
+
+        Console.Write($"Do you want to save logs to {logFile}? Y/N: ");
+
+        do
+        {
+            confirmationResult = Console.ReadKey();
+        }
+        while (confirmationResult.Key != ConsoleKey.Y && confirmationResult.Key != ConsoleKey.N);
+
+        if (confirmationResult.Key == ConsoleKey.Y)
+        {
+            WriteLogsToFile();
+            Console.WriteLine("\nLogs saved.");
+        }
+        else
+        {
+            Console.WriteLine("\nLogs will not be saved.");
+        }
```

Comment:
We can move thid to a separate method that will return true or false and name it as IsLogToFile(); this methid will check whether we need to save logs into files in this case we will be able to create smth like this: if(IsLogToFile()) { WriteLogsToFile();  }